src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,56 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunScenario(ProcessStartInfo processStartInfo, Action<PerformanceTestConfig> preIterationDel,
+            Action<PerformanceTestConfig> postIterationDel, Func<PerformanceTestConfig, ScenarioBenchmark> teardownDel)
+        {
+            int iterations = _performanceTestConfig.Iterations;
+            int timeout = (int)(_performanceTestConfig.TimeoutPerIteration.TotalMilliseconds);
+
+            for (int i = 0; i < iterations; i++)
+            {
+                preIterationDel(_performanceTestConfig);
+                using(var p = new Process())
+                {
+                    p.StartInfo = processStartInfo;
+                    p.Start();
+                    if (p.WaitForExit(timeout) == false) 
+                    {
+                        if (p != null)
+                        {
+                            p.Kill();
+                        }
+                        Console.WriteLine("Timeouted!");
+                        return;
+                    }
+                    postIterationDel(_performanceTestConfig);
+                }
+            }
+
+            ScenarioBenchmark scenarioBenchmark = teardownDel(_performanceTestConfig);
+            if (scenarioBenchmark == null) 
+            {
+                Console.Error.WriteLine ("The Teardown Delegate should return a valid instance of ScenarioBenchmark.");
+            }
+
+            string scenarioNamespace = scenarioBenchmark.Namespace;
+
+            scenarioBenchmark.Serialize(Configuration.RunId + "-" + scenarioNamespace + ".xml");
+
+            var mdFileName = Configuration.RunId + "-" + scenarioNamespace + "-Statistics.md";
+            var csvFileName = Configuration.RunId + "-" + scenarioNamespace + "-Statistics.csv";
+
+            var dt = scenarioBenchmark.GetStatistics();
+            var mdTable = MarkdownHelper.GenerateMarkdownTable(dt);
+            MarkdownHelper.Write(mdFileName, mdTable);
+            WriteInfoLine($"Markdown file saved to \"{mdFileName}\"");
+            Console.WriteLine(mdTable);
```

Comment:
So that doesn't seem to work. WriteInfoLine needs a string as a parameter, and mdTable is a MarkdownLog.Table.